client/post-editor/editor-sharing/publicize-services.jsx

```diff
@@ -1,59 +1,72 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { filter, uniqBy } from 'lodash';
 
 /**
  * Internal dependencies
  */
-var serviceConnections = require( 'my-sites/sharing/connections/service-connections' ),
-	EditorSharingPublicizeConnection = require( './publicize-connection' );
+import EditorSharingPublicizeConnection from './publicize-connection';
+import { getCurrentUserId } from 'state/current-user/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteUserConnections } from 'state/sharing/publicize/selectors';
 
-module.exports = React.createClass( {
-	displayName: 'EditorSharingPublicizeServices',
+class EditorSharingPublicizeServices extends Component {
+	static propTypes = {
+		post: PropTypes.object,
+		connections: PropTypes.array.isRequired,
+		newConnectionPopup: PropTypes.func.isRequired
+	};
 
-	propTypes: {
-		post: React.PropTypes.object,
-		siteId: React.PropTypes.number.isRequired,
-		connections: React.PropTypes.array.isRequired,
-		newConnectionPopup: React.PropTypes.func.isRequired
-	},
+	static defaultProps = {
+		post: Object.freeze( {} ),
+	};
 
-	renderServices: function() {
-		var services = serviceConnections.getServicesFromConnections( this.props.connections );
-
-		return services.map( function( service ) {
-			return (
-				<li key={ service.ID } className="editor-sharing__publicize-service">
-					<h5 className="editor-sharing__publicize-service-heading">{ service.label }</h5>
-					{ this.renderConnections( service.ID ) }
-				</li>
-			);
-		}, this );
-	},
+	renderServices() {
+		const services = uniqBy( this.props.connections.map( ( { label, service } ) => ( { label, service } ) ), 'service' );
 
-	renderConnections: function( serviceName ) {
-		const connections = serviceConnections.getConnectionsAvailableToCurrentUser(
-			serviceName,
-			this.props.connections
+		return services.map( ( { label, service } ) =>
+			<li key={ service } className="editor-sharing__publicize-service">
+				<h5 className="editor-sharing__publicize-service-heading">{ label }</h5>
+				{ this.renderConnections( service ) }
+			</li>
 		);
+	}
 
-		return connections.map( function( connection ) {
-			return (
-				<EditorSharingPublicizeConnection
-					key={ connection.ID }
-					post={ this.props.post }
-					connection={ connection }
-					onRefresh={ this.props.newConnectionPopup } />
-			);
-		}, this );
-	},
+	/**
+	 * Displays Publicize connections for the passed service.
+	 *
+	 * @param {String} service Slug of Keyring service.
+	 * @return {Object|DOMElement} React Element.
+	 */
+	renderConnections( service ) {
+		return filter( this.props.connections, { service } ).map( ( connection ) =>
+			<EditorSharingPublicizeConnection
+				key={ connection.ID }
+				post={ this.props.post }
+				connection={ connection }
+				onRefresh={ this.props.newConnectionPopup } />
+		);
+	}
 
-	render: function() {
+	render() {
 		return (
```

Comment:
I'm always in favor of consolidation, and I know you are too. Here I wondered how other reviewers would feel about loops two levels deep :)

Aside: What makes you prefer `_.map` over `[].map()`?
Aside2: What's the benefit of keying connections with their label instead of accessing that property directly within the loop?